tests/add_user.spec.ts

```diff
@@ -45,8 +52,10 @@ test.describe("Add User functional", () => {
     const addUserPage = new AddUserPage(page);
     const generator = new Generator();
 
-    await addUserPage.setGender(generator.getGender());
-    await addUserPage.setUserName(generator.getName(false), true);
+    const user = new UserDTO(generator.getGender(), generator.getName(false));
+
+    await addUserPage.setGender(user.getGender());
+    await addUserPage.setUserName(user.getUsername(), true);
```

Comment:
use universal fillUserFields, inside the fillUserFields check for undefined fields if you don want to fill so of the fields